packages/blocks/src/api/raw-handling/html-formatting-remover.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Internal dependencies
+ */
+import { isPhrasingContent } from './phrasing-content';
+
+function getSibling( node, which ) {
+	const sibling = node[ `${ which }Sibling` ];
+
+	if ( sibling ) {
+		return sibling;
+	}
+
+	const { parentNode } = node;
+
+	if ( ! parentNode ) {
+		return;
+	}
+
+	if ( ! isPhrasingContent( parentNode ) ) {
+		return;
+	}
+
+	return getSibling( parentNode, which );
+}
+
+function isFormattingSpace( character ) {
+	return character === ' ' || character === '\t' || character === '\n';
+}
+
+/**
+ * Removes spacing that formats HTML.
+ *
+ * @param {Node} node The node to be processed.
+ * @return {void}
+ */
+export default function( node ) {
+	if ( node.nodeType !== node.TEXT_NODE ) {
+		return;
+	}
+
+	// Ignore pre content.
+	if ( node.parentElement.closest( 'pre' ) ) {
+		return;
+	}
+
+	// First, replace any sequence of HTML formatting space with a single space.
+	let newData = node.data.replace( /[ \n\t]+/g, ' ' );
```

Comment:
Should we expected to need to account for carriage return `\r` ?

https://en.wikipedia.org/wiki/Carriage_return

Thinking is whether copied text could be formatted as `"Hello\r\nWorld"`

Reply:
Done.